core/src/main/scala/kafka/admin/ConfigCommand.scala

```diff
@@ -535,7 +537,7 @@ object ConfigCommand extends Config {
 
   class ConfigCommandOptions(args: Array[String]) extends CommandDefaultOptions(args) {
 
-    val zkConnectOpt = parser.accepts("zookeeper", "The connection string for the zookeeper connection in the form host:port. " +
+    val zkConnectOpt = parser.accepts("zookeeper", "DEPRECATED. The connection string for the zookeeper connection in the form host:port. " +
```

Comment:
Can add "Replaced by --bootstrap-server, REQUIRED unless --bootstrap-server is given." as per `PreferredReplicaLeaderElectionCommand`.  It might also be appropriate to add .requiredUnless("bootstrap-server") after parser.accepts() as long as bootstrap-server is defined in the parser before zookeeper, as per https://jopt-simple.github.io/jopt-simple/apidocs/joptsimple/OptionSpecBuilder.html#requiredUnless-java.lang.String-java.lang.String...-

```
parser.accepts("zookeeper", "DEPRECATED...").requiredUnless("bootstrap-server")
```

Reply:
Thanks for the suggestion @rondagostino, not very familiar with Scala so this is good to know :). Just so I'm clear here, the addition of `.requiredUnless("bootstrap-server")` would look something like this?
```
    val bootstrapServerOpt = parser.accepts("bootstrap-server", "The Kafka server to connect to. " +
      "This is required for describing and altering broker configs.")
      .withRequiredArg
      .describedAs("server to connect to")
      .ofType(classOf[String])
    val zkConnectOpt = parser.accepts("zookeeper", "DEPRECATED. The connection string for the zookeeper connection in the form host:port. " +
      "Multiple URLS can be given to allow fail-over. Replaced by --bootstrap-server, REQUIRED unless --bootstrap-server is given.")
      .requiredUnless("bootstrap-server")
      .withRequiredArg
      .describedAs("urls")
      .ofType(classOf[String])
```

Reply:
I agree that it's a good idea to suggest using --bootstrap-server instead.  I'm not sure we need to use `requireUnless`, since I think that the argument parsing code already handles this

Reply:
@cmccabe is correct, in the code here: https://github.com/apache/kafka/blob/8a7db5a595d4acd515aedfe118bb1205f2356f9f/core/src/main/scala/kafka/admin/ConfigCommand.scala#L663-L666 It checks that only 1 option is specified, and if 0 or both options are specified, it throws the appropriate exception. In this case then it might be redundant to have a `requireUnless` here. I will update the help statement as suggested and push.

Reply:
@rondagostino There also seems to be a test for invalid args (if neither boostrap servers or zk is specified here:

https://github.com/apache/kafka/blob/8a7db5a595d4acd515aedfe118bb1205f2356f9f/core/src/test/scala/unit/kafka/admin/ConfigCommandTest.scala#L48-L50